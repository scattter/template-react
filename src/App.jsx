import React, {useState} from 'react'

const App = () => {
  const [count, setCount] = useState(0)
  
  return (
    <>
      <h1>APP</h1>
      <h2>qwe</h2>
      <span>count is {count}</span>
      <button onClick={() => setCount((c) => c + 1)}>add</button>
    </>
  )
}

export default App
