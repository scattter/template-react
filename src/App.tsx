import React, { useState } from 'react';
import '@/index.scss';
import { Demo } from './constant';
import { Home } from './pages/home';

const App = () => {
  const [count, setCount] = useState(0);

  return (
    <>
      <h1>APP demo 3 5 7</h1>
      <h2>{Demo.first}</h2>
      <span>count is {count}</span>
      <button className="button" onClick={() => setCount((c) => c + 1)}>
        add
      </button>
      <Home />
    </>
  );
};

export default App;
