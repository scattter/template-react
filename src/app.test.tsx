import { shallow, ShallowWrapper } from 'enzyme';

import App from './App';

describe('test App.tsx', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should success increment count when click button', async () => {
    const span = wrapper.find('span').first();
    expect(span.text()).toEqual('count is 0');
    wrapper.find('button').first().simulate('click');
    await (() => {
      expect(span.text()).toEqual('count is 1');
    });
  });
});
