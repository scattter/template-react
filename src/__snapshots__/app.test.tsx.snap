// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test App.tsx snapshot test 1`] = `
<Fragment>
  <h1>
    APP demo 3 5 7
  </h1>
  <h2>
    1
  </h2>
  <span>
    count is 
    0
  </span>
  <button
    className="button"
    onClick={[Function]}
  >
    add
  </button>
  <Home />
</Fragment>
`;
